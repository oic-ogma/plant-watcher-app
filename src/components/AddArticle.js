import React, { Component } from 'react';
import { View } from 'react-native';
import { Label, FormInputComponent } from './common';
import { FormValidationMessage, Button } from 'react-native-elements';

export default class AddArticle extends Component {
  saveArticle() {
    const { plantName, articleContents, saveArticle } = this.props;
    saveArticle(plantName, articleContents);
  }

  render() {
    const {
      plantName,
      articleContents,
      plantNameChanged,
      articleContentsChanged,
      error
    } = this.props;

    return (
      <View>
        <Label placeholder='植物名' />
        <FormInputComponent
          onChangeText={plantNameChanged}
          placeholder='ひまわり'
          value={plantName}
        />

        <Label placeholder='記事'/>
        <FormInputComponent
          placeholder='黄色い'
          onChangeText={articleContentsChanged}
          multiline={true}
          value={articleContents}
        />
        <FormValidationMessage labelStyle={{ fontSize: 16 }}>{error}</FormValidationMessage>

        <Button
          onPress={this.saveArticle.bind(this)}
          title='投稿'
          raised
          large
          loading={this.props.loading}
          backgroundColor='green'
        />

      </View>
    );
  }
}
