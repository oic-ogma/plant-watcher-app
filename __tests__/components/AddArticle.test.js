import React from 'react';
import AddArticle from '../../src/components/AddArticle';
import renderer from 'react-test-renderer';

describe('記事投稿', () => {
  it('正しくrenderできる', () => {
    const tree = renderer.create(<AddArticle/>).toJSON();
    expect(tree).toBeTruthy();
  });

  it('snapshot-普通', () => {
    const tree = renderer.create(<AddArticle/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot-植物名と記事が入力されている', () => {
    const INITIAL_STATE = {
      plantName: 'ひまわり',
      articleContents: 'きいろ\nまるい',
      loading: false,
      error: ''
    };
    const tree = renderer.create(<AddArticle {...INITIAL_STATE}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot-ロード中', () => {
    const INITIAL_STATE = {
      plantName: 'ひまわり',
      articleContents: 'きいろ\nまるい',
      loading: true,
      error: ''
    };
    const tree = renderer.create(<AddArticle {...INITIAL_STATE}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot-エラーが出ている', () => {
    const INITIAL_STATE = {
      plantName: 'ひまわり',
      articleContents: 'きいろ\nまるい',
      loading: false,
      error: 'hogehoge'
    };
    const tree = renderer.create(<AddArticle {...INITIAL_STATE}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
