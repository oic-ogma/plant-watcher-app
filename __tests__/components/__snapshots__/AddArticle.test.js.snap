// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`記事投稿 snapshot-エラーが出ている 1`] = `
<View
  style={
    Object {
      "borderBottomWidth": 0,
      "borderColor": "#ddd",
      "borderRadius": 2,
      "borderWidth": 1,
      "elevation": 1,
      "marginLeft": 5,
      "marginRight": 5,
      "marginTop": 10,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": 40,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 3,
            "fontSize": 18,
            "paddingLeft": 10,
          }
        }
      >
        植物名
      </Text>
      <TextInput
        autoCorrect={false}
        multiline={undefined}
        onChangeText={[Function]}
        placeholder="ひまわり"
        secureTextEntry={undefined}
        style={
          Object {
            "color": "#000",
            "flex": 5,
            "fontSize": 18,
            "lineHeight": 23,
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
        underlineColorAndroid="transparent"
        value="ひまわり"
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "height": 40,
          },
          Object {
            "height": 120,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 3,
            "fontSize": 18,
            "paddingLeft": 10,
          }
        }
      >
        記事
      </Text>
      <TextInput
        autoCorrect={false}
        multiline={true}
        onChangeText={[Function]}
        placeholder="植物についての説明"
        secureTextEntry={undefined}
        style={
          Object {
            "color": "#000",
            "flex": 5,
            "fontSize": 18,
            "lineHeight": 23,
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
        underlineColorAndroid="transparent"
        value={undefined}
      />
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "#FF0000",
        "fontSize": 18,
        "paddingTop": 5,
      }
    }
  >
    hogehoge
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#fff",
          "borderColor": "#007aff",
          "borderRadius": 5,
          "borderWidth": 1,
          "flex": 1,
          "marginLeft": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "#007aff",
            "fontSize": 18,
            "fontWeight": "600",
            "paddingBottom": 15,
            "paddingTop": 15,
          }
        }
      >
        投稿
      </Text>
    </View>
  </View>
</View>
`;

exports[`記事投稿 snapshot-ロード中 1`] = `
<View
  style={
    Object {
      "borderBottomWidth": 0,
      "borderColor": "#ddd",
      "borderRadius": 2,
      "borderWidth": 1,
      "elevation": 1,
      "marginLeft": 5,
      "marginRight": 5,
      "marginTop": 10,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": 40,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 3,
            "fontSize": 18,
            "paddingLeft": 10,
          }
        }
      >
        植物名
      </Text>
      <TextInput
        autoCorrect={false}
        multiline={undefined}
        onChangeText={[Function]}
        placeholder="ひまわり"
        secureTextEntry={undefined}
        style={
          Object {
            "color": "#000",
            "flex": 5,
            "fontSize": 18,
            "lineHeight": 23,
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
        underlineColorAndroid="transparent"
        value="ひまわり"
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "height": 40,
          },
          Object {
            "height": 120,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 3,
            "fontSize": 18,
            "paddingLeft": 10,
          }
        }
      >
        記事
      </Text>
      <TextInput
        autoCorrect={false}
        multiline={true}
        onChangeText={[Function]}
        placeholder="植物についての説明"
        secureTextEntry={undefined}
        style={
          Object {
            "color": "#000",
            "flex": 5,
            "fontSize": 18,
            "lineHeight": 23,
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
        underlineColorAndroid="transparent"
        value={undefined}
      />
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "#FF0000",
        "fontSize": 18,
        "paddingTop": 5,
      }
    }
  >
    
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#999999"
        hidesWhenStopped={true}
        size="large"
      />
    </View>
  </View>
</View>
`;

exports[`記事投稿 snapshot-普通 1`] = `
<View
  style={
    Object {
      "borderBottomWidth": 0,
      "borderColor": "#ddd",
      "borderRadius": 2,
      "borderWidth": 1,
      "elevation": 1,
      "marginLeft": 5,
      "marginRight": 5,
      "marginTop": 10,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": 40,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 3,
            "fontSize": 18,
            "paddingLeft": 10,
          }
        }
      >
        植物名
      </Text>
      <TextInput
        autoCorrect={false}
        multiline={undefined}
        onChangeText={[Function]}
        placeholder="ひまわり"
        secureTextEntry={undefined}
        style={
          Object {
            "color": "#000",
            "flex": 5,
            "fontSize": 18,
            "lineHeight": 23,
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
        underlineColorAndroid="transparent"
        value={undefined}
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "height": 40,
          },
          Object {
            "height": 120,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 3,
            "fontSize": 18,
            "paddingLeft": 10,
          }
        }
      >
        記事
      </Text>
      <TextInput
        autoCorrect={false}
        multiline={true}
        onChangeText={[Function]}
        placeholder="植物についての説明"
        secureTextEntry={undefined}
        style={
          Object {
            "color": "#000",
            "flex": 5,
            "fontSize": 18,
            "lineHeight": 23,
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
        underlineColorAndroid="transparent"
        value={undefined}
      />
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "#FF0000",
        "fontSize": 18,
        "paddingTop": 5,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#fff",
          "borderColor": "#007aff",
          "borderRadius": 5,
          "borderWidth": 1,
          "flex": 1,
          "marginLeft": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "#007aff",
            "fontSize": 18,
            "fontWeight": "600",
            "paddingBottom": 15,
            "paddingTop": 15,
          }
        }
      >
        投稿
      </Text>
    </View>
  </View>
</View>
`;

exports[`記事投稿 snapshot-植物名と記事が入力されている 1`] = `
<View
  style={
    Object {
      "borderBottomWidth": 0,
      "borderColor": "#ddd",
      "borderRadius": 2,
      "borderWidth": 1,
      "elevation": 1,
      "marginLeft": 5,
      "marginRight": 5,
      "marginTop": 10,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "height": 40,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 3,
            "fontSize": 18,
            "paddingLeft": 10,
          }
        }
      >
        植物名
      </Text>
      <TextInput
        autoCorrect={false}
        multiline={undefined}
        onChangeText={[Function]}
        placeholder="ひまわり"
        secureTextEntry={undefined}
        style={
          Object {
            "color": "#000",
            "flex": 5,
            "fontSize": 18,
            "lineHeight": 23,
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
        underlineColorAndroid="transparent"
        value="ひまわり"
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "height": 40,
          },
          Object {
            "height": 120,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 3,
            "fontSize": 18,
            "paddingLeft": 10,
          }
        }
      >
        記事
      </Text>
      <TextInput
        autoCorrect={false}
        multiline={true}
        onChangeText={[Function]}
        placeholder="植物についての説明"
        secureTextEntry={undefined}
        style={
          Object {
            "color": "#000",
            "flex": 5,
            "fontSize": 18,
            "lineHeight": 23,
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
        underlineColorAndroid="transparent"
        value={undefined}
      />
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
        "color": "#FF0000",
        "fontSize": 18,
        "paddingTop": 5,
      }
    }
  >
    
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 1,
          "borderColor": "#ddd",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 5,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#fff",
          "borderColor": "#007aff",
          "borderRadius": 5,
          "borderWidth": 1,
          "flex": 1,
          "marginLeft": 5,
          "marginRight": 5,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "#007aff",
            "fontSize": 18,
            "fontWeight": "600",
            "paddingBottom": 15,
            "paddingTop": 15,
          }
        }
      >
        投稿
      </Text>
    </View>
  </View>
</View>
`;
